src/Microsoft.ML.LightGBM.StaticPipe/LightGbmStaticExtensions.cs

```diff
@@ -93,17 +129,56 @@ public static class LightGbmStaticExtensions
             int? numLeaves = null,
             int? minDataPerLeaf = null,
             double? learningRate = null,
-            int numBoostRound = LightGbmArguments.Defaults.NumBoostRound,
-            Action<LightGbmArguments> advancedSettings = null,
+            int numBoostRound = Options.Defaults.NumBoostRound,
             Action<IPredictorWithFeatureWeights<float>> onFit = null)
         {
-            CheckUserValues(label, features, weights, numLeaves, minDataPerLeaf, learningRate, numBoostRound, advancedSettings, onFit);
+            CheckUserValues(label, features, weights, numLeaves, minDataPerLeaf, learningRate, numBoostRound, onFit);
 
             var rec = new TrainerEstimatorReconciler.BinaryClassifier(
                (env, labelName, featuresName, weightsName) =>
                {
                    var trainer = new LightGbmBinaryTrainer(env, labelName, featuresName, weightsName, numLeaves,
-                       minDataPerLeaf, learningRate, numBoostRound, advancedSettings);
+                       minDataPerLeaf, learningRate, numBoostRound);
+
+                   if (onFit != null)
+                       return trainer.WithOnFitDelegate(trans => onFit(trans.Model));
+                   else
+                       return trainer;
+               }, label, features, weights);
+
+            return rec.Output;
+        }
+
+        /// <summary>
+        /// Predict a target using a tree binary classification model trained with the <see cref="LightGbmBinaryTrainer"/>.
+        /// </summary>
+        /// <param name="catalog">The <see cref="BinaryClassificationCatalog"/>.</param>
+        /// <param name="label">The label column.</param>
+        /// <param name="features">The features column.</param>
+        /// <param name="weights">The weights column.</param>
+        /// <param name="options">Algorithm advanced settings.</param>
+        /// <param name="onFit">A delegate that is called every time the
+        /// <see cref="Estimator{TInShape, TOutShape, TTransformer}.Fit(DataView{TInShape})"/> method is called on the
+        /// <see cref="Estimator{TInShape, TOutShape, TTransformer}"/> instance created out of this. This delegate will receive
+        /// the linear model that was trained. Note that this action cannot change the result in any way;
+        /// it is only a way for the caller to be informed about what was learnt.</param>
+        /// <returns>The set of output columns including in order the predicted binary classification score (which will range
+        /// from negative to positive infinity), the calibrated prediction (from 0 to 1), and the predicted label.</returns>
+        public static (Scalar<float> score, Scalar<float> probability, Scalar<bool> predictedLabel) LightGbm(this BinaryClassificationCatalog.BinaryClassificationTrainers catalog,
+            Scalar<bool> label, Vector<float> features, Scalar<float> weights,
```

Comment:
Ditto "`options` API" comment.

Reply:
these are static extensions  (details in other comment)

---
In reply to: [251533977](https://github.com/dotnet/machinelearning/pull/2261#discussion_r251533977) [](ancestors = 251533977)